backend/layers/persistence/persistence.py

```diff
@@ -104,13 +104,19 @@ def _get_serializable_session(self, **kwargs):
             yield session
 
     def _row_to_collection_version(self, row: Any, canonical_collection: CanonicalCollection) -> CollectionVersion:
+        if isinstance(row.collection_metadata, CollectionMetadata) or row.collection_metadata is None:
```

Comment:
It would be worth pulling this out into a helper function since we are using it in `_row_to_collection_version_with_datasets` and `_row_to_collection_version`